pkg/types/datum.go

```diff
@@ -2082,6 +2082,16 @@ func (d *Datum) ToString() (string, error) {
 	}
 }
 
+// StringWithCtx implements Explainable interface.
+func (d *Datum) StringWithCtx(ctx ParamValues, redact string) string {
+	switch d.Kind() {
+	case KindVectorFloat32:
+		return d.GetVectorFloat32().StringWithCtx(ctx, redact)
+	default:
+		return fmt.Sprintf("%v", d.GetValue())
```

Comment:
It should support the redact mode. Please add the test cases to test vector type whether to support the redact mode.

Reply:
Here I just `return d.GetVectorFloat32().StringWithCtx(ctx, redact) `.
Process redact mode in `vector.go::StringWithRedact()`
Some test has been add to test `redact=off, on, marker`